tests/serverpod_test_server/test_integration/authentication/auth_config_changes_test.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:serverpod_auth_server/module.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() async {
+  var userName = 'test';
+  var email = 'test@serverpod.dev';
+  var password = 'password';
+  var session = await IntegrationTestServer().session();
+
+  var codeLengthsToTest = [-16, -8, -4, -1, 0, 1, 4, 16];
+
+  test(
+      'Not setting validationCodeLength will generate a code with default length 8',
+      () async {
+    String? generatedValidationCode;
+    AuthConfig.set(
+      AuthConfig(
+        sendValidationEmail: (session, email, validationCode) async {
+          generatedValidationCode = validationCode;
+          return true;
+        },
+      ),
+    );
+
+    var createAccountRequest =
+        await Emails.createAccountRequest(session, userName, email, password);
+
+    expect(
+      createAccountRequest,
+      isTrue,
+      reason: 'Generated validation code is 8 characters long, which is valid',
+    );
+
+    assert(generatedValidationCode?.length == 8);
+  });
+
+  for (var length in codeLengthsToTest) {
+    final willPass = length >= 1;
+    test(
+        willPass
+            ? 'Setting validationCodeLength to $length will generate a code with the given length'
+            : 'Setting validationCodeLength to $length will return null',
+        () async {
+      String? generatedValidationCode;
+      AuthConfig.set(
+        AuthConfig(
+          validationCodeLength: length,
+          sendValidationEmail: (session, email, validationCode) async {
+            generatedValidationCode = validationCode;
+            return true;
+          },
+        ),
+      );
```

Comment:
Discussion: If we move the throw into the constructor we can expect that the AuthConfig constructor throws directly instead of having to rely on the `set` method and call `createAccountRequest`. 

You can see some example code for expecting exceptions here if you are not familiar with it from before: https://github.com/serverpod/serverpod/blob/a29ecdb8221020afc433f5b94612620d08607d89/packages/serverpod/test/database/table_relation_test.dart#L9-L19

Reply:
Yes this is a much better approach! Thanks :)  

Just to make sure: For the test cases that are not supposed to throw an Exception can I just check for:

`expect(() => AuthConfig(validationCodeLength: 4), 4);`

Or should i still call `createAccountRequest` there?